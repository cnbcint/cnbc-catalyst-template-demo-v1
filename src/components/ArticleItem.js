import React, { useContext } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { LayoutContext } from "./Main";

const Container = styled.div`
  a {
    text-decoration: none;
    color: ${(props) => props.layout.colors.copy};

    h5 {
      font-weight: normal;
      font-size: 1.2rem;
    }
    &:hover {
      h5 {
        color: ${(props) => props.layout.colors.primary};
      }
      span {
        color: ${(props) => props.layout.colors.copy};
      }
    }
    span {
      font-size: 1rem;
      font-weight: bold;
      text-transform: uppercase;
      color: ${(props) => props.layout.colors.primary};
    }
    img {
      margin: 0;
      margin-bottom: 1rem;
      width: 100%;
    }
    p {
      font-size: 1rem;
      line-height: 1.2rem;
      opacity: 0.75;
    }
  }
`;

export default function ArticleItem(props) {
  const layout = useContext(LayoutContext);
  return (
    <Container className={props.className} layout={layout}>
      <a href={props.link} target={props.target}>
        <img src={props.image} loading="lazy" alt="Thumbnail" />
        {props.tagline && <span>{props.tagline}</span>}
        <h5>{props.title}</h5>
        {props.copy && <p>{props.copy}</p>}
      </a>
    </Container>
  );
}
ArticleItem.defaultProps = {
  target: "_self",
  className: "",
};
ArticleItem.propTypes = {
  className: PropTypes.string,
  link: PropTypes.string.isRequired,
  target: PropTypes.oneOf(["_self", "_blank"]),
  image: PropTypes.string.isRequired,
  tagline: PropTypes.string,
  copy: PropTypes.string,
  title: PropTypes.string.isRequired,
};
