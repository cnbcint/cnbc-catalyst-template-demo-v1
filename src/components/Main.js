import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { getMargin, breakpoints } from "../helpers/Layout";
import { ModalContext } from "../components/Modal";

export const LayoutContext = React.createContext({
  maxWidth: "500px",
  colors: null,
});

const Container = styled.div`
  display: block;
  width: 100%;
  background-color: ${(props) => props.colors.background};
  > * {
    font-size: 1.125rem;
    line-height: 140%;
    color: ${(props) => props.colors.copy};
  }
  hr {
    // border-color: ${(props) => props.primaryColor};
    border-color: ${(props) => props.colors.copy};
    border-style: solid;
    display: inline-block;
    width: 100%;
    // margin-top: ${getMargin(1)};
    margin-bottom: ${getMargin(1)};
    @media (min-width: ${breakpoints.mobile}) {
      // margin-top: ${getMargin(2)};
      margin-bottom: ${getMargin(2)};
    }
  }
  img {
    width: 100%;
    display: block;
    margin-top: ${getMargin(2)};
    margin-bottom: ${getMargin(2)};
  }
  p {
    margin: 1rem auto !important;
  }
  ul {
    list-style: disc !important;
    margin-top: ${getMargin(1)} !important;
    margin-bottom: ${getMargin(1)} !important;
    li {
      list-style: disc !important;
      margin: ${getMargin(1)} !important;
      margin-left: ${getMargin(2)} !important;
    }
  }
`;
export default function Main(props) {
  return (
    <LayoutContext.Provider
      value={{ maxWidth: props.maxWidth, colors: props.colors }}
    >
      <ModalContext.Provider value={{ modals: {} }}>
        <Container
          colors={props.colors}
          primaryColor={props.colors.primary}
          className={props.className}
          style={props.style}
        >
          {props.children}
        </Container>
      </ModalContext.Provider>
    </LayoutContext.Provider>
  );
}
Main.defaultProps = {
  maxWidth: "900px",
  colors: { primary: "black", secondary: "gray", tertiary: "red" },
};
Main.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
  maxWidth: PropTypes.string,
  colors: PropTypes.shape({
    primary: PropTypes.string,
    secondary: PropTypes.string,
    tertiary: PropTypes.string,
    background: PropTypes.string,
    copy: PropTypes.string,
  }),
  style: PropTypes.object,
};
