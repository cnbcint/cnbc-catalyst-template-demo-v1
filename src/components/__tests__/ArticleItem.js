import React from "react";
import ArticleItem from "../ArticleItem";
import Main from "../Main";
import { render } from "@testing-library/react";
import "jest-styled-components";

const Theme = {
  maxWidth: "900px",
  colors: {
    primary: "orange",
    secondary: "green",
    tertiary: "orange",
    background: "white",
    copy: "black",
    links: "blue",
  },
};
const testProps = {
  title: "Test Title",
  copy: "Test copy",
  image: "images/test.jpg",
  link: "http://www.test.com",
};

describe("ArticleItem", () => {
  it("Renders correctly", () => {
    const { container } = render(
      <Main {...Theme}>
        <ArticleItem {...testProps} />
      </Main>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("has an img with the correct source", () => {
    const { container } = render(
      <Main {...Theme}>
        <ArticleItem {...testProps} />
      </Main>
    );
    expect(container.querySelector("img").src).toMatch(
      "http://localhost/images/test.jpg"
    );
  });
  it("has an img with the correct title", () => {
    const { container } = render(
      <Main {...Theme}>
        <ArticleItem {...testProps} />
      </Main>
    );
    expect(container.querySelector("h5").textContent).toMatch("Test Title");
  });
  it("has an img with the correct copy", () => {
    const { container } = render(
      <Main {...Theme}>
        <ArticleItem {...testProps} />
      </Main>
    );
    expect(container.querySelector("p").textContent).toMatch("Test copy");
  });

  it("has an img with the correct link", () => {
    const { container } = render(
      <Main {...Theme}>
        <ArticleItem {...testProps} />
      </Main>
    );
    expect(container.querySelector("a").href).toMatch("http://www.test.com");
  });
});
