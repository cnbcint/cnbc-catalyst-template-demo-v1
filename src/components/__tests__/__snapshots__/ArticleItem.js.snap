// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleItem Renders correctly 1`] = `
.c0 {
  display: block;
  width: 100%;
  background-color: white;
}

.c0 > * {
  font-size: 1.125rem;
  line-height: 140%;
  color: black;
}

.c0 hr {
  border-color: black;
  border-style: solid;
  display: inline-block;
  width: 100%;
  margin-bottom: 1rem;
}

.c0 img {
  width: 100%;
  display: block;
  margin-top: 2rem;
  margin-bottom: 2rem;
}

.c0 p {
  margin-top: 1rem !important;
  margin-bottom: 1rem !important;
}

.c0 ul {
  list-style: disc !important;
  margin-top: 1rem !important;
  margin-bottom: 1rem !important;
}

.c0 ul li {
  list-style: disc !important;
  margin: 1rem !important;
  margin-left: 2rem !important;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
}

.c1 a h5 {
  font-weight: normal;
  font-size: 1.2rem;
}

.c1 a:hover h5 {
  color: orange;
}

.c1 a:hover span {
  color: black;
}

.c1 a span {
  font-size: 1rem;
  font-weight: bold;
  text-transform: uppercase;
  color: orange;
}

.c1 a img {
  margin: 0;
  margin-bottom: 1rem;
  width: 100%;
}

.c1 a p {
  font-size: 1rem;
  line-height: 1.2rem;
  opacity: 0.75;
}

@media (min-width:768px) {
  .c0 hr {
    margin-bottom: 2rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <a
      href="http://www.test.com"
      target="_self"
    >
      <img
        alt="Thumbnail"
        loading="lazy"
        src="images/test.jpg"
      />
      <h5>
        Test Title
      </h5>
      <p>
        Test copy
      </p>
    </a>
  </div>
</div>
`;
